src/middlewared/middlewared/plugins/rsync.py

```diff
@@ -310,7 +310,7 @@ def handle_ssh_dir():
                         try:
                             async with await asyncssh.connect(
                                 **connect_kwargs,
-                                known_hosts=known_hosts,
+                                known_hosts=known_hosts if not data['ssh_credentials'] else None,
```

Comment:
We need to specify known_hosts in any case so that we verify that they are valid